File: Two Sum/1_Two_Sum.go
func twoSum(nums []int, target int) []int {
	var result []int

	var mp map[int]int
	mp = make(map[int]int)

	for i := 0; i < len(nums); i++ {
		mp[nums[i]] = i + 1
	}

	var x, y, xPos, yPos int
	for i := 0; i < len(nums); i++ {
		x = nums[i]
		y = target - x
		xPos = i
		yPos = mp[y] - 1

		if mp[y] > 0 && xPos != yPos {
			break
		}
	}

	result = append(result, xPos, yPos)

	return result
}

File: Two Sum/1_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int len = nums.size();
        int i, left, right;
        map<int, int> mp;
        for( i=0 ; i<len ; i++ ){
            mp[nums[i]] = i+1;
        }
        
        for( i=0 ; i<len ; i++ ){
            left = nums[i];
            right = target - nums[i];
            if(mp[right] > 0){
                left = i;
                right = mp[right] - 1;
                if(left != right) break;
            }
        }
        
        vector<int> res;
        res.push_back(left);
        res.push_back(right);
        return res;
    }
};
